packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+	 */
+	const onOptionSelect: DropdownProps["onOptionSelect"] = (event, data) => {
+		setType(data.optionText ?? "undefined");
+		setEditingComponent(selectEditUI(data.optionText ?? "undefined"));
+		if (data.optionText === "undefined") {
+			submitChange(undefined);
+		}
+
+		if (data.optionText === " null") {
+			// We need to support users waiting to use "null" as a value
+			// eslint-disable-next-line unicorn/no-null
+			submitChange(null);
+		}
+	};
+
+	const options = node.editProps?.editTypes;
+	return options === undefined ? (
+		<TreeHeader
+			label={label}
+			nodeTypeMetadata={node.typeMetadata}
+			inlineValue={String(node.value)}
+		/>
+	) : (
+		<div
+			style={
+				editing
+					? {
+							border: "2px solid",
+							borderColor: tokens.colorBrandStroke1,
+							display: "flex",
+					  }
+					: { display: "flex" }
+			}
+			onBlur={onBlur}
+			onFocus={onFocus}
+			onClick={(event): void => event?.preventDefault()}
+		>
+			<TreeHeader label={label} nodeTypeMetadata={node.typeMetadata} />
+			<Dropdown
+				size="small"
+				style={{ minWidth: "10px" }}
+				onOptionSelect={onOptionSelect}
+				value={type}
+				selectedOptions={[type]}
+			>
+				{options.map((option) => (
+					<Option key={option}>{option}</Option>
+				))}
+			</Dropdown>
+			{editingComponent}
+		</div>
+	);
+}
+
+/**
+ * Input to {@link EditBooleanComponent}
+ */
+interface EditableBooleanComponentProps {
+	node: FluidObjectValueNode;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for boolean editing
+ */
+function EditableBooleanComponent(props: EditableBooleanComponentProps): React.ReactElement {
+	const { node, submitChange } = props;
+
+	/**
+	 * Auxiliary function to make boolean editing buttons
+	 * @param isTrue - Determines if it should be a true or false button
+	 * @returns - 2 buttons, one for setting value to false and one for true
+	 */
+	function boolButton(isTrue: boolean): React.ReactElement {
+		return (
+			<Button
+				size="small"
+				disabled={isTrue === node.value}
+				onClick={(event): void => {
+					event.preventDefault();
+					event.stopPropagation();
+					submitChange(isTrue);
+				}}
+			>
+				{String(isTrue).charAt(0).toUpperCase() + String(isTrue).slice(1)}
+			</Button>
+		);
+	}
+	return (
+		<>
+			{boolButton(true)}
+			{boolButton(false)}
+		</>
+	);
+}
+
+/**
+ * Input to {@link EditTextComponent}
+ */
+interface EditableTextComponentProps {
+	node: FluidObjectValueNode;
+	setEditing: React.Dispatch<React.SetStateAction<boolean>>;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for text editing
+ */
+function EditableTextComponent(props: EditableTextComponentProps): React.ReactElement {
+	const { node, setEditing, submitChange } = props;
+	const [localData, setLocalData] = React.useState<string>(
+		typeof node.value !== "string" ? "" : node.value,
```

Comment:
Probably worth documenting that we are intentionally clearing out the form contents if the current node value's type is not string for UX reasons.